### ISSUE TITLE
Add Support For Heatmap

### ISSUE THREAD

**Issue by zolrath** [istarkov/google-map-react#122]
It would be nice to have support for the Google Maps Heatmap Layer.
https://developers.google.com/maps/documentation/javascript/examples/layer-heatmap

The ability to pass in an array of {lat, long, weight} objects into a heatmap component would be great!

Or is this out of scope for this component?

**Comment by istarkov** [comment]
As you have access to internal map object in api callback, you could add any google layer. All thing other than just render react components are out of the scope now ;-). For similar things I create HOC components over GoogleMap component and use features I need.

**Comment by ruta-goomba** [comment]
An example of heatmap layer implementation:

 ```
<GoogleMap
        defaultCenter={default_position}
        defaultZoom={6}
        yesIWantToUseGoogleMapApiInternals
        onGoogleApiLoaded={({map, maps}) => {
          console.log(points[0]);
          const heatmap = new maps.visualization.HeatmapLayer({
            data: points.map(point => (
              {location: new maps.LatLng(point['location'][1], point['location'][0]),
              weight: point['weight']}))
          });
          heatmap.setMap(map);
        }}
      >
```

where points is of the following structure:

```
[
{location:[-1.131592, 52.629729], weight: 2},
{location:[-1.141592, 52.629729], weight: 3},
{location:[-1.161592, 53.629729], weight: 1},
...
]
```

**Comment by whollacsek** [comment]
@ruta-goomba in your example `maps.visualization` is undefined, could you explain how you did it?

**Comment by jackzampolin** [comment]
bump

**Comment by whollacsek** [comment]
@jackzampolin what I did is adding: 
```
<GoogleMap
      bootstrapURLKeys={{
        libraries: 'visualization',
      }}
>
...
```
Then `maps.visualization.HeatmapLayer` will be available.

**Comment by jackzampolin** [comment]
@whollacsek Thank you for getting me unblocked on that!

**Comment by awatson1978** [comment]
Thank you much for the Heatmap hints!  Got it working!

**Comment by whollacsek** [comment]
You're welcome :)

Le jeu. 18 mai 2017 à 06:32, Abigail Watson <notifications@github.com> a
écrit :

**Comment by aaroncrawford** [comment]
FYI for those in the future, the example above by ruta works, however the lat and lng are reversed in the point.map function.  Should be location[0], then location [1] if you stick to the LatLng naming schema.

**Comment by rushikeshpandit22** [comment]
heat map layer location is not exactly  as given.
![heatmap](https://user-images.githubusercontent.com/14141622/29652875-1d3d0f3c-88c5-11e7-84ce-0a9343229335.png)

**Comment by echowand** [comment]
FYI for future users, don't forget visualization lib. 
src="https://maps.googleapis.com/maps/api/js?key=YOUR_API_KEY**&libraries=visualization**"

**Comment by rushikeshpandit22** [comment]
Yes, I specified 'libraries=visualization' in URL-It's display heat map but the issue is it not on the exact  location on zoom-in and when we do zoom-out it's view gets split.
